### PR TITLE
Move httpx requirement, Fix Ails and ChatgptAi Provider, Improve scripts

### PR DESCRIPTION
**Pull request by hlohaus**

### CODE REVIEW

**Comment by xtekky** [comment]
looks like it has some conflicts could you fix them ?

**Comment by hlohaus** [comment]
I have fixed the conflicts.

**Comment by johnd0e** [comment]
Acytoo streaming:
> ...
You can try our new always-free website: https://chat.acytoo.com
Or telegram bot @chatacytoo_bot
For developers, you can use https://api.acytoo.net directly.Cancelling an overlapped future failed
future: <_OverlappedFuture pending overlapped=<pending, 0x234816ed120> cb=[_ProactorReadPipeTransport._loop_reading()]>
Traceback (most recent call last):
  File "C:\Users\anton\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 71, in _cancel_overlapped
    self._ov.cancel()
OSError: [WinError 6] The handle is invalid

**Comment by johnd0e** [comment]
HuggingChat streaming:
> ...Cancelling an overlapped future failed
future: <_OverlappedFuture pending overlapped=<pending, 0x29639acd1b0> cb=[_ProactorReadPipeTransport._loop_reading()]>
Traceback (most recent call last):
  File "C:\Users\anton\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 71, in _cancel_overlapped
    self._ov.cancel()
OSError: [WinError 6] The handle is invalid

**Comment by johnd0e** [comment]
Liaobots:
> Traceback (most recent call last):
  File "D:\AI\gpt4free\stream.py", line 21, in <module>
    for message in response:
  File "D:\AI\gpt4free\g4f\Provider\base_provider.py", line 93, in create_completion
    yield from run_generator(cls.create_async_generator(model, messages, stream=stream, **kwargs))
  File "D:\AI\gpt4free\g4f\Provider\base_provider.py", line 122, in run_generator
    yield loop.run_until_complete(gen.__anext__())
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "D:\AI\gpt4free\g4f\Provider\Liaobots.py", line 64, in create_async_generator
    response.raise_for_status()
  File "D:\AI\gpt4free\venv\lib\site-packages\aiohttp\client_reqrep.py", line 1005, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url=URL('https://liaobots.com/api/user')

**Comment by johnd0e** [comment]
Vercel:
```
D:\AI\gpt4free\venv\lib\site-packages\curl_cffi\requests\session.py:471: UserWarning:
WindowsProactorEventLoopPolicy is not supported, you can use the selector loop by:
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

  warnings.warn(WINDOWS_WARN)
Exception ignored from cffi callback <function socket_function at 0x000002493A18F1C0>:
Traceback (most recent call last):
  File "D:\AI\gpt4free\venv\lib\site-packages\curl_cffi\aio.py", line 62, in socket_function
    loop.add_reader(sockfd, async_curl.process_data, sockfd, CURL_CSELECT_IN)
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 518, in add_reader
    raise NotImplementedError
NotImplementedError:
Exception ignored from cffi callback <function socket_function at 0x000002493A18F1C0>:
Traceback (most recent call last):
  File "D:\AI\gpt4free\venv\lib\site-packages\curl_cffi\aio.py", line 68, in socket_function
    async_curl._sockfds.remove(sockfd)
KeyError: 812
Exception ignored from cffi callback <function socket_function at 0x000002493A18F1C0>:
Traceback (most recent call last):
  File "D:\AI\gpt4free\venv\lib\site-packages\curl_cffi\aio.py", line 65, in socket_function
    loop.add_writer(sockfd, async_curl.process_data, sockfd, CURL_CSELECT_OUT)
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 524, in add_writer
    raise NotImplementedError
NotImplementedError:
Exception ignored from cffi callback <function socket_function at 0x000002493A18F1C0>:
Traceback (most recent call last):
  File "D:\AI\gpt4free\venv\lib\site-packages\curl_cffi\aio.py", line 62, in socket_function
    loop.add_reader(sockfd, async_curl.process_data, sockfd, CURL_CSELECT_IN)
  File "C:\Users\a\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 518, in add_reader
    raise NotImplementedError
NotImplementedError:
```

File: g4f/Provider/Aichat.py [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-from ..typing import Any, CreateResult
-from .base_provider import BaseProvider
+from .base_provider import AsyncProvider, format_prompt
 
 
-class Aichat(BaseProvider):
+class Aichat(AsyncProvider):
     url                   = "https://chat-gpt.org/chat"
     working               = True
     supports_gpt_35_turbo = True
```
[the remark this answers is not in the document]
Comment (johnd0e) [reply g4f/Provider/Aichat.py:12]: Also these also do not work with "stream" argument:
- Bing https://github.com/xtekky/gpt4free/commit/5ca47b44b2b42abb4f48163c17500b5ee67ab28f#r126756961
- ChatgptLogin https://github.com/xtekky/gpt4free/commit/5ca47b44b2b42abb4f48163c17500b5ee67ab28f#r126757292
- OpenaiChat https://github.com/xtekky/gpt4free/commit/5ca47b44b2b42abb4f48163c17500b5ee67ab28f#r126757515
- Vercel https://github.com/xtekky/gpt4free/commit/5ca47b44b2b42abb4f48163c17500b5ee67ab28f#r126757946
- Wewordle https://github.com/xtekky/gpt4free/commit/5ca47b44b2b42abb4f48163c17500b5ee67ab28f#r126758335

File: g4f/Provider/Vercel.py [hunk 2]
```diff
@@ -1,63 +1,72 @@
...
-from .base_provider import BaseProvider
 
-
-class Vercel(BaseProvider):
-    url                   = "https://play.vercel.ai"
+class Vercel(AsyncProvider):
+    url                   = "https://sdk.vercel.ai"
     working               = True
     supports_gpt_35_turbo = True
+    model                 = "replicate:replicate/llama-2-70b-chat"
```
Comment (johnd0e) [inline g4f/Provider/Vercel.py:14]: Not in `models.py`

**Comment by johnd0e** [comment]
@hlohaus

**Comment by hlohaus** [comment]
Try Not use Windows?

**Comment by johnd0e** [comment]
I have Windows only.
But it should not be hard to setup some sample @ replit.com.
Do you want me to do it?

File: g4f/Provider/Vercel.py (hunk 2 above)
Comment (hlohaus) [reply g4f/Provider/Vercel.py:14]: I a added "g4f.models.default" for HuggingChat and OpenAssistant.

File: g4f/Provider/Aichat.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hlohaus) [reply g4f/Provider/Aichat.py:12]: My fault. I added "supports_stream" only in AsyncGeneratorProvider.
Try to use "create_completion" or "create_async_generator" function from the provider.

**Comment by hlohaus** [comment]
The windows error is from the "Vercel" example.
But it is only a warning. Can you disable display warnings?

Liabots is currently down.
For HuggingChat you need cookies.

Try to use this script:
python testing/test_async.py
It should not have the windows issue and it show you all working providers.

**Pull request merged**